components/jenkins/keyvault.tf

```
@@ -26,3 +27,24 @@ resource "azurerm_key_vault_secret" "db" {
   value        = azurerm_cosmosdb_account.cosmosdb.primary_master_key
   key_vault_id = azurerm_key_vault.jenkinskv.id
 }
+
+resource "random_password" "jenkins-agent-password" {
+  length  = 16
+  special = true
+  lower   = true
+  upper   = true
+  number  = true
+}
+
+resource "azurerm_key_vault_secret" "jenkins-agent-password" {
+  name         = "jenkins-agent-password"
+  value        = random_password.jenkins-agent-password
+  key_vault_id = azurerm_key_vault.jenkinskv.id
+}
+
+resource "azurerm_key_vault_secret" "subscription_id" {
+  name         = "subscription_id"
```

isn't this subscription-id?

```suggestion
  name         = "subscription-id"
```

---

ah yeah